helix-view/src/editor.rs

```diff
@@ -146,10 +146,22 @@ pub struct Config {
     pub lsp: LspConfig,
 }
 
-#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 pub struct LspConfig {
+    /// Display LSP progress messages below statusline
     pub display_messages: bool,
+    /// Enable automatic pop up of signature help (parameter hints)
+    pub auto_signature_help: bool,
+}
+
+impl Default for LspConfig {
+    fn default() -> Self {
+        Self {
+            display_messages: false,
+            auto_signature_help: true,
+        }
+    }
```

Comment:
The defaulting here behaves strangely if you have a config with one or the other filled in, like:

```toml
[editor.lsp]
display-messages = true
```

I think because the Default here applies to the whole LspConfig. Is there a way to default the fields individually?

Reply:
Yeah, https://serde.rs/attr-default.html

Reply:
Fixed in 485d18cc.